File: Universal Windows Platform Solution/Phantom of the West/Phantom of the West/Visual Novel/Chapters/Chapter 1/Chapter1.cs
using System.Collections.Generic;

namespace Phantom_of_the_West.Visual_Novel.Chapters.Chapter_1
{
	internal class Chapter1 : Chapter
	{
		internal static Dictionary<int, StoryChoiceListDelegate> dictionary = new Dictionary<int, StoryChoiceListDelegate>
		{
			{1001, Ch1Title},
			{1002, Ch1IAmDrKaden},
			{1003, Ch1ThisIsNewFerando},
			{1004, Ch1IDrinkElixir},
			{1005, Ch1ILeaveHotel},
			{1006, Ch1IGoToAlley},
			{1007, Ch1OfficerSeesMe},
			{1008, Ch1MyFaceDisappears},
			{1009, Ch1IOpenMyEyes},
			{1010, Ch1IPossessOfficer},
			{1011, Ch1ILookAtMyHands},
			{1012, Ch1IGoToOfficersHome},
			{1013, Ch1ISleep},
		};

		internal Chapter1() : base(dictionary)
		{
		}

		private static StoryChoiceList Ch1Title()
		{
			return StoryChoiceList.FromResource("Black.png", new IStoryChoice[] { StoryChoice.FromResource("ch1Title", 1002) });
		}

		private static StoryChoiceList Ch1IAmDrKaden()
		{
			return StoryChoiceList.FromResource("I-am-Dr-Kaden.png", new IStoryChoice[] { StoryChoice.FromResource("ch1IAmDrKaden", 1003) });
		}

		private static StoryChoiceList Ch1ThisIsNewFerando()
		{
			return StoryChoiceList.FromResource("Cityscape-from-Window.png", new IStoryChoice[] { StoryChoice.FromResource("ch1ThisIsNewFerando", 1004) });
		}

		private static StoryChoiceList Ch1IDrinkElixir()
		{
			return StoryChoiceList.FromResource("Dr-Kaden-Drinks-Elixir.png", new IStoryChoice[] { StoryChoice.FromResource("ch1IDrinkElixir", 1005) });
		}

		private static StoryChoiceList Ch1ILeaveHotel()
		{
			return StoryChoiceList.FromResource("Dr-Kaden-Walking-among-Citizens.png", new IStoryChoice[] { StoryChoice.FromResource("ch1ILeaveHotel", 1006) });
		}

		private static StoryChoiceList Ch1IGoToAlley()
		{
			return StoryChoiceList.FromResource("Dr-Kaden-Ill.png", new IStoryChoice[] { StoryChoice.FromResource("ch1IGoToAlley", 1007) });
		}

		private static StoryChoiceList Ch1OfficerSeesMe()
		{
			return StoryChoiceList.FromResource("Officer-Notices-Dr-Kaden.png", new IStoryChoice[] { StoryChoice.FromResource("ch1OfficerSeesMe", 1008) });
		}

		private static StoryChoiceList Ch1MyFaceDisappears()
		{
			return StoryChoiceList.FromResource("Phantom-Reveal.png", new IStoryChoice[] { StoryChoice.FromResource("ch1MyFaceDisappears", 1009) });
		}

		private static StoryChoiceList Ch1IOpenMyEyes()
		{
			return StoryChoiceList.FromResource("Phantom-Reveal-with-Eyes.png", new IStoryChoice[] { StoryChoice.FromResource("ch1IOpenMyEyes", 1010) });
		}

		private static StoryChoiceList Ch1IPossessOfficer()
		{
			return StoryChoiceList.FromResource("Possess-Police-Officer.png", new IStoryChoice[] { StoryChoice.FromResource("ch1IPossessOfficer", 1011) });
		}

		private static StoryChoiceList Ch1ILookAtMyHands()
		{
			return StoryChoiceList.FromResource("Officer-Hands.png", new IStoryChoice[] { StoryChoice.FromResource("ch1ILookAtMyHands", 1012) });
		}

		private static StoryChoiceList Ch1IGoToOfficersHome()
		{
			return StoryChoiceList.FromResource("Drive-to-Officer-Home.png", new IStoryChoice[] { StoryChoice.FromResource("ch1IGoToOfficersHome", 1013) });
		}

		private static StoryChoiceList Ch1ISleep()
		{
			return StoryChoiceList.FromResource("Officer-on-Bed.png", new IStoryChoice[] { StoryChoice.FromResource("ch1ISleep", 2001) });
		}
	}
}

File: Android Project/app/src/main/java/com/christopherraleigh/phantomofthewest/visual_novel/chapters/chapter_1/Chapter1.java
package com.christopherraleigh.phantomofthewest.visual_novel.chapters.chapter_1;

import com.christopherraleigh.phantomofthewest.R;
import com.christopherraleigh.phantomofthewest.visual_novel.IStoryChoice;
import com.christopherraleigh.phantomofthewest.visual_novel.IStoryChoiceList;
import com.christopherraleigh.phantomofthewest.visual_novel.chapters.Chapter;
import com.christopherraleigh.phantomofthewest.visual_novel.chapters.StoryChoice;
import com.christopherraleigh.phantomofthewest.visual_novel.chapters.StoryChoiceList;

import java.util.Arrays;

/**
 * Created by christopherraleigh on 2016-05-11.
 */
public class Chapter1 extends Chapter {
    private static StoryChoiceList ch1Title = new StoryChoiceList(R.drawable.black,
            Arrays.asList(new IStoryChoice[]{new StoryChoice(R.string.ch1Title, 1002)}));

    private static StoryChoiceList ch1IAmDrKaden = new StoryChoiceList(R.drawable.i_am_dr_kaden,
            Arrays.asList(new IStoryChoice[]{new StoryChoice(R.string.ch1IAmDrKaden, 1003)}));

    private static StoryChoiceList ch1ThisIsNewFerando = new StoryChoiceList(R.drawable.cityscape_from_window,
            Arrays.asList(new IStoryChoice[]{new StoryChoice(R.string.ch1ThisIsNewFerando, 1004)}));

    private static StoryChoiceList ch1IDrinkElixir = new StoryChoiceList(R.drawable.dr_kaden_drinks_elixir,
            Arrays.asList(new IStoryChoice[]{new StoryChoice(R.string.ch1IDrinkElixir, 1005)}));

    private static StoryChoiceList ch1ILeaveHotel = new StoryChoiceList(R.drawable.dr_kaden_walking_among_citizens,
            Arrays.asList(new IStoryChoice[]{new StoryChoice(R.string.ch1ILeaveHotel, 1006)}));

    private static StoryChoiceList ch1IGoToAlley = new StoryChoiceList(R.drawable.dr_kaden_ill,
            Arrays.asList(new IStoryChoice[]{new StoryChoice(R.string.ch1IGoToAlley, 1007)}));

    private static StoryChoiceList ch1OfficerSeesMe = new StoryChoiceList(R.drawable.officer_notices_dr_kaden,
            Arrays.asList(new IStoryChoice[]{new StoryChoice(R.string.ch1OfficerSeesMe, 1008)}));

    private static StoryChoiceList ch1MyFaceDisappears = new StoryChoiceList(R.drawable.phantom_reveal,
            Arrays.asList(new IStoryChoice[]{new StoryChoice(R.string.ch1MyFaceDisappears, 1009)}));

    private static StoryChoiceList ch1IOpenMyEyes = new StoryChoiceList(R.drawable.phantom_reveal_with_eyes,
            Arrays.asList(new IStoryChoice[]{new StoryChoice(R.string.ch1IOpenMyEyes, 1010)}));

    private static StoryChoiceList ch1IPossessOfficer = new StoryChoiceList(R.drawable.possess_police_officer,
            Arrays.asList(new IStoryChoice[]{new StoryChoice(R.string.ch1IPossessOfficer, 1011)}));

    private static StoryChoiceList ch1ILookAtMyHands = new StoryChoiceList(R.drawable.officer_hands,
            Arrays.asList(new IStoryChoice[]{new StoryChoice(R.string.ch1ILookAtMyHands, 1012)}));

    private static StoryChoiceList ch1IGoToOfficersHome = new StoryChoiceList(R.drawable.drive_to_officer_home,
            Arrays.asList(new IStoryChoice[]{new StoryChoice(R.string.ch1IGoToOfficersHome, 1013)}));

    private static StoryChoiceList ch1ISleep = new StoryChoiceList(R.drawable.officer_on_bed,
            Arrays.asList(new IStoryChoice[]{new StoryChoice(R.string.ch1ISleep, 2001)}));

    @Override
    public IStoryChoiceList goToState(int id) {
        switch (id) {
            case 1001:
                return ch1Title;
            case 1002:
                return ch1IAmDrKaden;
            case 1003:
                return ch1ThisIsNewFerando;
            case 1004:
                return ch1IDrinkElixir;
            case 1005:
                return ch1ILeaveHotel;
            case 1006:
                return ch1IGoToAlley;
            case 1007:
                return ch1OfficerSeesMe;
            case 1008:
                return ch1MyFaceDisappears;
            case 1009:
                return ch1IOpenMyEyes;
            case 1010:
                return ch1IPossessOfficer;
            case 1011:
                return ch1ILookAtMyHands;
            case 1012:
                return ch1IGoToOfficersHome;
            case 1013:
                return ch1ISleep;
        }
        return null;
    }
}
